src/pykoala/instruments/koala_ifu.py

```diff
@@ -858,163 +866,908 @@ def get_rss_mask(rss, **kwargs):   # THIS TASK SHOULD BE A CORRECTION CLASS, inc
+    if brightest_fibres_to_combine is None: brightest_fibres_to_combine = 10
+    if lowest_fibres_to_combine is None: lowest_fibres_to_combine = 10
+    fmax = np.nanmedian(rss.intensity[rss.koala.integrated_fibre_sorted[-brightest_fibres_to_combine:]], axis=0)
+    fmin = np.nanmedian(rss.intensity[rss.koala.integrated_fibre_sorted[:lowest_fibres_to_combine]], axis=0)    
+    s = fmax-fmin                                       
+    index_max = s.tolist().index(np.nanmax(s))
+    w_max = rss.wavelength[index_max] 
+    if verbose: print("> Quick checking wavelength of the brightest emission line in this rss:", w_max)
+    return (w_max)
+
+
+def apply_mask_to_rss(rss,              # THIS TASK SHOULD BE A CORRECTION CLASS, including apply() #TODO
+                      mask = None,   
+                      make_zeros=False, 
+                      **kwargs):    
+    """
+    Apply a mask to a RSS. 
 
+    Parameters
+    ----------
+    rss : object
+        rss object where the mask will be applied.
+    mask : np.array, optional
+        mask. The default is None.
+    make_zeros : boolean, optional
+        When applying mask, put 0s instead of nans. The default is False.
+    **kwargs : kwargs
+        where we can find verbose, verbose_counter, warnings, plot...
 
+    Returns
+    -------
+    rss_out : 
+        rss with the mask applied.
+
+    """
+    verbose = kwargs.get('verbose', False)
+    rss_out = copy.deepcopy(rss)
+    mask_in_rss = False
+    if mask is not None:
+        mask_provided = True                                   # Mask provided 
+        if str(type(mask)) == "<class 'str'>":   # It is a file, read mask TODO
+            pass           
+    else:
+        mask_provided = False         # Mask NOT provided 
+        try:
+            if rss.mask is not None: 
+                mask = rss.mask
+                mask_in_rss = True
+        except Exception:
+            mask = get_rss_mask(rss_out)
+    rss_out.intensity = rss_out.intensity * mask
+    rss_out.variance = rss_out.variance * mask        
+    rss_out.mask = np.array(mask)
+    if verbose: 
+        if mask_provided:
+            print("> Mask provided has been applied to rss.") 
+        else:
+            if mask_in_rss: 
+                if make_zeros:
+                    print("> Mask included in rss applied to make 0 all bad pixels in edges.")
+                else:
+                    print("> Mask included in rss applied to make nan all bad pixels in edges.")    
+            else:
+                if make_zeros:
+                    print("> Mask computed and applied to make 0 all bad pixels in edges.")
+                else:
+                    print("> Mask computed and applied to make nan all bad pixels in edges.")  
+    return rss_out
+
+
+def clean_nan(rss, **kwargs): #TODO: THIS TASK SHOULD BE A CORRECTION CLASS, including apply()
+    """
+    Clean nans in rss using interpolation (this is needed to avoid 1D tasks failing )
```

Comment:
I agree this is indeed a correction.

On the other hand, all tasks in `onedspec` should cope with the `nan`s (and/or any other `DataMask`), shouldn't they?